### ISSUE TITLE
[FR/QST] How do I flush stdout in a `%%script bash` cell?

### ISSUE THREAD

**Issue by NightMachinery** [jupyter/notebook#6277]
Prescript: I posted this first on the Discourse, but there is some module active there that bans IPs from VPS subnets, which I use to escape Iran's Internet filtering. I suggest using some other antispam measure.

---

How do I flush stdout in a `%%script bash` cell in a Jupyter notebook (Google Colab, actually)?

I have some long-running script cells, and I like to see the results as they are coming out, not waiting for the whole cell to finish.

**Comment by meeseeksmachine** [comment]
This issue has been mentioned on **Jupyter Community Forum**. There might be relevant details there:

https://discourse.jupyter.org/t/how-do-i-flush-stdout-in-a-script-bash-cell/12699/5

**Comment by Matoran** [comment]
```
%%script bash
echo hi ; sleep 5
```
This seems to output hi and then sleep 5 seconds. Issue solved?

**Comment by krassowski** [comment]
Yes, this was improved in IPython 8.32 by https://github.com/ipython/ipython/pull/14695, closing as solved (though it never was a notebook issue, it was always an IPython issue).

**Comment by Matoran** [comment]
@krassowski thanks for the feedback, following issues are related and can also be closed :).
https://github.com/jupyter/notebook/issues/4048
https://github.com/jupyter/notebook/issues/6291

**Comment by krassowski** [comment]
Thank you @Matoran!
